src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -89,29 +90,48 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         }
     }
 
-    private fun checkUnsafeCall(holder: AnnotationHolder, o: RsExpr) {
-        val parent = o.ancestors
-            .drop(1)    // skip the expression itself
+    private fun PsiElement.isInUnsafeBlockOrFn(parentsToSkip: Int = 0): Boolean {
+        val parent = this.ancestors
+            .drop(parentsToSkip)
             .find {
                 when (it) {
                     is RsBlockExpr -> it.unsafe != null
                     is RsFunction -> true
                     else -> false
                 }
-            } ?: return
+            } ?: return false
+
+        return parent is RsBlockExpr || (parent is RsFunction && parent.unsafe != null)
+    }
 
-        if (parent is RsBlock || (parent is RsFunction && parent.unsafe == null)) {
+    private fun checkUnsafeCall(holder: AnnotationHolder, o: RsExpr) {
+        if (!o.isInUnsafeBlockOrFn(/* skip the expression itself*/ 1)) {
             val annotation = holder.createErrorAnnotation(o, "Call to unsafe function requires unsafe function or block [E0133]")
             annotation.registerFix(SurroundWithUnsafeFix(o))
             val block = o.parentOfType<RsBlock>()?.parent ?: return
             annotation.registerFix(AddUnsafeFix(block))
         }
     }
 
+    private fun checkUnsafePtrDereference(holder: AnnotationHolder, o: RsUnaryExpr) {
+        val type = o.expr?.type
+        if (!(type is TyPointer))
+            return
+
+        if (!o.isInUnsafeBlockOrFn()) {
+            val annotation = holder.createErrorAnnotation(o, "Dereference of raw pointer requires unsafe function or block [E0133]")
+            annotation.registerFix(SurroundWithUnsafeFix(o))
+            val block = o.parentOfType<RsBlock>()?.parent ?: return
+            annotation.registerFix(AddUnsafeFix(block))
+        }
+    }
+
     private fun checkUnaryExpr(holder: AnnotationHolder, unaryExpr: RsUnaryExpr) {
         val expr = unaryExpr.expr ?: return
         if (unaryExpr.operatorType == UnaryOperator.REF_MUT && !expr.isMutable()) {
             createImmutableErrorAnnotation(holder, expr)
+        } else if (unaryExpr.operatorType == UnaryOperator.DEREF) {
+            checkUnsafePtrDereference(holder, unaryExpr)
```

Comment:
There is already a SurroundWithUnsafeFix and a AddUnsafeFix being added on line 123-125.
Or am i misunderstanding your request?